### PR TITLE
Spelling fixes and improved explanations of toString, toSQL, and as.

### PR DESCRIPTION
**Pull request by hanneswidrig**
I just started out using Knex.js, really enjoying using it but I found some clarification problems within the documentation. Most notably in the toString() explanation.

### CODE REVIEW

File: sections/builder.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default [
     type: "method",
     method: "as",
     example: ".as(name)",
-    description: "Allows for aliasing a subquery, taking the string you wish to name the current query. If the query is not a sub-query, it will be ignored.",
+    description: "Allows for aliasing a subquery, taking the string you wish to name the current query. NOTE: Does not create SQL alias for columns or tables.",
```
Comment (elhigu) [inline sections/builder.js]: I don't understand what you mean by this addition / change.

File: sections/interfaces.js [hunk 2]
```diff
@@ -368,9 +368,17 @@ export default [
     description: "Returns an array of query strings filled out with the correct values based on bindings, etc. Useful for debugging.",
     children: [
       {
-        type: "runnable",
+        type: "code",
+        language: "js",
         content: `
-          knex.select('*').from('users').where(knex.raw('id = ?', [1])).toString()
+          var toStringQuery = knex.select('*').from('users').where('id', 1).toString();
+          console.log(toStringQuery.toString());
```
Comment (elhigu) [inline sections/interfaces.js]: you are calling toString() for string here?

File: sections/interfaces.js [hunk 3]
```diff
@@ -368,9 +368,17 @@ export default [
...
+          
+          // toString() must be called on the variable object to print out query to console.
+          // Outputs:
+
+          select * from "users" where id = 1
```
Comment (elhigu) [inline sections/interfaces.js]: this should be in comment, since its not javascript code

Comment (elhigu) [reply sections/interfaces.js]: `// Outputs: select * from "users" where id = 1`

File: sections/builder.js (hunk 1 above)
Comment (hanneswidrig) [reply sections/builder.js]: I feel the documentation should be more clear that the as() function only works on sub-queries.

Comment (hanneswidrig) [reply sections/builder.js]: fair enough, I don't really know how to word this better. I guess its fine if we just leave it as is.

File: sections/interfaces.js (hunk 2 above)
Comment (hanneswidrig) [reply sections/interfaces.js]: If I remove the .toString() function call from ```toStringQuery```, will it still print out the query?

File: sections/interfaces.js (hunk 3 above)
Comment (hanneswidrig) [reply sections/interfaces.js]: Okay interesting my bad, I had a ton of dependency trouble trying to actually run the documentation on a local node.js server.

**Comment by elhigu** [comment]
I think this is fine now, thanks!

**Pull request merged**